rest-server/core/src/main/scala/com/actionml/core/search/elasticsearch/ElasticSearchSupport.scala

```diff
@@ -225,76 +226,66 @@ class ElasticSearchClient private (alias: String, client: RestClient)(implicit w
     p.future
   }
 
-  private def performRequest(method: String, url: String): Future[Response] = {
-    Future(client.performRequest(new Request(method, url)))
-  }
-
-  def findDocById(id: String): EsDocument = {
-    var rjv: Option[JValue] = None
-    try {
-      client.performRequest(new Request("HEAD", s"/_alias/$alias")) // Does the alias exist?
-        .getStatusLine
-        .getStatusCode match {
-        case 200 =>
-          val aliasResponse = client.performRequest(new Request("GET", s"/_alias/$alias"))
-          val responseJValue = parse(EntityUtils.toString(aliasResponse.getEntity))
-          val indexSet = responseJValue.extract[Map[String, JValue]].keys
-          rjv = indexSet.headOption.map { actualIndexName =>
-            val url = s"/$actualIndexName/$indexType/${encodeURIFragment(id)}"
-            logger.trace(s"Find doc by id using URL: $url")
-            val response = client.performRequest(new Request("GET", url))
-            logger.trace(s"Got response: $response")
+  def findDocById(id: String): EsDocument = Await.result(findDocByIdAsync(id), Duration.Inf)
```

Comment:
@andreyshikov Wouldn't a timeout be better?

Reply:
> @andreyshikov Wouldn't a timeout be better? I mean Duration.Inf seems pretty long :-)

future will fail on at least ESclient timeout, and this timeout setting will be kind of duplication. I think it's not needed here...